include/libtorrent/proxy_base.hpp

```diff
@@ -41,11 +41,11 @@ POSSIBILITY OF SUCH DAMAGE.
 
 namespace libtorrent {
 
-class proxy_base : boost::noncopyable
+class proxy_base
 {
 public:
 
-	typedef std::function<void(error_code const&)> handler_type;
+	using handler_type = std::function<void(error_code const&)>;
```

Comment:
this should probably be reviewed. with c++11 the handler should probably be moved instead of heap allocated